ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -277,20 +304,19 @@ public void onMessage(final Packet packet, final DiscoveryPeer sender) {
       return;
     }
 
-    if (!isPeerPermitted(sender, localPeer)) {
-      LOG.trace("Dropping packet from peer not in the whitelist ({})", sender.getEnodeURLString());
+    if (!isPeerPermittedToSendMessages(sender)) {
+      LOG.trace("Dropping packet from disallowed peer ({})", sender.getEnodeURLString());
       return;
     }
 
     // Load the peer from the table, or use the instance that comes in.
     final Optional<DiscoveryPeer> maybeKnownPeer = peerTable.get(sender);
     final DiscoveryPeer peer = maybeKnownPeer.orElse(sender);
     final boolean peerKnown = maybeKnownPeer.isPresent();
-    final boolean peerBlacklisted = peerBlacklist.contains(peer);
```

Comment:
Without the blacklist peers that are "bad" will still hit our peer table.  Is this acceptable per what we need to do?  Will we advertise that we've seen "blackbeard" even if we refuse to talk to him?  Should we be tracking peer-no-grata?  Or are we checking permissions at every time we would pull said peer out of the peer table?

Reply:
The blacklist basically transformed into `peerPermissions` which gets checked in the method  `isPeerPermittedToSendMessages` above on line 307.  So we're still filtering out the bad peers here.